app/services/users/replace_mentions_service.rb

```diff
@@ -0,0 +1,162 @@
+#  OpenProject is an open source project management software.
+#  Copyright (C) 2010-2022 the OpenProject GmbH
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License version 3.
+#
+#  OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+#  Copyright (C) 2006-2013 Jean-Philippe Lang
+#  Copyright (C) 2010-2013 the ChiliProject Team
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+#  See COPYRIGHT and LICENSE files for more details.
+
+# Rewrites mentions in user provided text (e.g. work package journals) from one user to another.
+# No data is to be removed.
+module Users
+  class ReplaceMentionsService
+    include ActiveRecord::Sanitization
+
+    def self.replacements
+      [
+        { class: WorkPackage, column: :description },
+        { class: Journal::WorkPackageJournal, column: :description },
+        { class: Document, column: :description },
+        { class: Journal, column: :notes },
+        { class: Comment, column: :comments },
+        { class: CustomValue, column: :value },
```

Comment:
As a mention could only happen in text custom value, the number of executions on this could be reduced to just the matching field_format. This would mean customizing the SQL however, or providing a way to join custom_fields.

Reply:
I added a condition to that replacement definition so that now only `text` custom values will be eligible for replacement.